services/api/src/models/user.js

```diff
@@ -16,6 +16,22 @@ schema.virtual('password').set(function (password) {
   this._password = password;
 });
 
+schema.methods.addAuthToken = function ({ exp, jti, iat }, ctx) {
```

Comment:
I would prefer not to tightly couple the model methods to the routes context. In addition to knock on effects of this coupling (it's unlikely but not impossible that a job could create users) it also makes testing more complicated (you have some code for this below).

Also is it standard to record the IP and userAgent in the token payload?

Reply:
there is things call ip bound jwt, but i dont think we are there yet.

Reply:
I do kind of like it... I feel it's common to see an IP attached to login sessions in other sites.
One thing however, wouldn't this IP be subject to change and need to be updated? On the assumption that it's defined as the "last known IP" then it would change if for example the users moved to a different wifi network.

`userAgent` seems likewise useful... I was just wondering if it was commonplace to store it on a JWT token... in theory the user agent wouldn't update as it would be tied to the device... I suppose it's not impossible though

Reply:
ha yeah i was thinking about this last night, as i was consider removing it.
I did an update to fetchUser let me know what you think about that.

we could store the userAgent but what would the benefit be ?

Reply:
I expect it would be useful to identify what device is being used for which token, the end goal being "IP: xxx, Device: iPhone, [Logout From This Device]"

Reply:
for IP if we're going to store it then I think `/users/me` is the appropriate place to update... ie:

```
if (ctx.get('ip') !== token.ip) {
   token.ip = ctx.get('ip');
   await user.save();
} 
```

etc